sensor/kubernetes/fake/flows.go

```diff
@@ -120,6 +121,30 @@ func generateIP() string {
 	return fmt.Sprintf("10.%d.%d.%d", rand.Intn(256), rand.Intn(256), rand.Intn(256))
 }
 
+func isPublicIP(ip string) bool {
+	return net.ParseIP(ip).IsPublic()
+}
+
+func generateExternalIP() string {
+	for {
+		ip := fmt.Sprintf("%d.%d.%d.%d", rand.Intn(256), rand.Intn(256), rand.Intn(256), rand.Intn(256))
+
+		if isPublicIP(ip) {
+			return ip
+		}
+	}
+}
+
+// We want to reuse some external IPs, so we test the cases where muliple
+// entities connect to the same external IP, but we also want many external IPs
+// that are only used once.
+func (w *WorkloadManager) generateExternalIPPool() {
+	for range 1000 {
+		ip := generateExternalIP()
+		externalIpPool.add(ip)
+	}
```

Comment:
There is a risk that we generate duplicate IPs - is this ok or do we want a firm 1000 unique IPs at this point? The side effect of duplicates is that they'll be more likely to be selected later on, but I think that's probably fine?

An alternative approach could be to explicitly generate the first thousand IPs in the subnet and then just use those, and generate a random one for anything "new" selected later on

Reply:
I think it is fine to have slightly fewer than 1000 unique IPs. It is an arbitrary number and I don't think it matters if it varies slightly from test to test. Given that there are more than three billion public IP addresses the probability that there is a collision is not very high. I think you would need to pick around 60,000 to have a good chance of a collision.